### PR TITLE
Adding some error handling and updates to the code

### PR DESCRIPTION
**Pull request by quadewarren**

### CODE REVIEW

File: library/artifactory_repo.py [hunk 1]
```diff
@@ -232,17 +239,9 @@
...
-
-# Update a virtual repository and register current config after update.
-- name: update test-virtual-update repo
-  artifactory_repo:
-    artifactory_url: https://artifactory.repo.example.com
-    auth_token: your_token
-    repo: "test-virtual-update"
+    name: "test-virtual-update"
     state: present
+    repo_config_dict:
```
Comment (gregswift) [inline library/artifactory_repo.py]: you didn't add any cofigs under the dict

File: library/artifactory_repo.py [hunk 2]
```diff
@@ -232,17 +239,9 @@
...
-
-# Update a virtual repository and register current config after update.
-- name: update test-virtual-update repo
-  artifactory_repo:
-    artifactory_url: https://artifactory.repo.example.com
-    auth_token: your_token
-    repo: "test-virtual-update"
+    name: test-virtual-update
     state: present
+    repo_config_dict:
```
Comment (gregswift) [inline library/artifactory_repo.py]: still didnt add anything here :P

File: library/artifactory_repo.py [hunk 3]
```diff
@@ -254,15 +253,21 @@
...
   artifactory_repo:
     artifactory_url: https://artifactory.repo.example.com
     auth_token: your_token
-    repo: "test-virtual-update"
+    name: test-virtual-update
     state: present
     repo_config_dict:
       description: "New public description"
-      url: "http://new.url.example.com"
+      url: http://new.url.example.com
```
Comment (gregswift) [inline library/artifactory_repo.py]: why does a virtual have a url?

Comment (quadewarren) [reply library/artifactory_repo.py]: Copy/pasta artifact, will fix.

File: library/artifactory_repo.py (hunk 2 above)
Comment (quadewarren) [reply library/artifactory_repo.py]: This line actually needs to be removed. The example is two tasks down.

**Review by gregswift — APPROVED** [review]

**Pull request merged**